secantTests/AppTests/AppInitializationTests.swift

```diff
@@ -82,56 +82,52 @@ class AppInitializationTests: XCTestCase {
             sandboxState: .placeholder,
             welcomeState: .placeholder
         )
-        
+
         let store = TestStore(
             initialState: appState,
             reducer: AppReducer()
         )
-        
+
         store.dependencies.databaseFiles = .noOp
         store.dependencies.databaseFiles.areDbFilesPresentFor = { _ in true }
-        store.dependencies.derivationTool = .noOp
+        store.dependencies.derivationTool = .liveValue
         store.dependencies.mainQueue = testScheduler.eraseToAnyScheduler()
         store.dependencies.mnemonic = .mock
         store.dependencies.randomRecoveryPhrase = recoveryPhraseRandomizer
         store.dependencies.walletStorage.exportWallet = { .placeholder }
         store.dependencies.walletStorage.areKeysPresent = { true }
 
         // Root of the test, the app finished the launch process and triggers the checks and initializations.
-        store.send(.appDelegate(.didFinishLaunching))
-        
+        _ = await store.send(.appDelegate(.didFinishLaunching))
+
         // the 0.02 delay ensures keychain is ready
-        testScheduler.advance(by: 0.02)
-        
+        await testScheduler.advance(by: 0.02)
+
         // ad 1.
-        store.receive(.checkWalletInitialization)
+        await store.receive(.checkWalletInitialization)
 
         // ad 2.
-        store.receive(.respondToWalletInitializationState(.initialized))
+        await store.receive(.respondToWalletInitializationState(.initialized))
 
         // ad 3.
-        store.receive(.initializeSDK) { state in
-            state.storedWallet = StoredWallet(
-                language: .english,
-                seedPhrase: "",
-                version: 0,
-                birthday: 0,
-                hasUserPassedPhraseBackupTest: false
-            )
+        await store.receive(.initializeSDK) { state in
+            state.storedWallet = .placeholder
         }
         // ad 4.
-        store.receive(.checkBackupPhraseValidation) { state in
+        await store.receive(.checkBackupPhraseValidation) { state in
             state.appInitializationState = .initialized
         }
 
         // the 3.0 delay ensures the welcome screen is visible till the initialization is done
-        testScheduler.advance(by: 3.00)
+        await testScheduler.advance(by: 3.00)
 
         // ad 5.
-        store.receive(.updateRoute(.phraseDisplay)) { state in
+        await store.receive(.updateRoute(.phraseDisplay)) { state in
             state.prevRoute = .welcome
             state.internalRoute = .phraseDisplay
         }
+
+        await store.finish(timeout: NSEC_PER_SEC * 5)
```

Comment:
Why do we wait 5 seconds instead of waiting on all in-flight effects to finish? Normally it just works with 
```
await store.finish()
```